test/proxy/utils/UUPSUpgradeable.test.js

```diff
@@ -6,6 +6,8 @@ const UUPSUpgradeableUnsafeMock = artifacts.require('UUPSUpgradeableUnsafeMock')
 const UUPSUpgradeableBrokenMock = artifacts.require('UUPSUpgradeableBrokenMock');
 const CountersImpl = artifacts.require('CountersImpl');
 
+const Legacy = [ './legacy/UUPSUpgradeableMockV1' ];
```

Comment:
Re: Testing. As I've said before, I think we need to test agains the source code. What it seems we need is to simply copy the current code for `_upgradeToAndCallSecure`, along with the definition of `_setImplementation` since it's private, into a testing contract.